handlers/eth1Account.go

```diff
@@ -166,6 +169,15 @@ func Eth1Address(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	code, err := eth1data.GetCodeAt(ctx, common.BytesToAddress(addressBytes))
+	if err != nil {
+		logger.WithError(err).Errorf("error retrieving code data for address %v", address)
+	}
+	isContract := len(code) != 0
+
```

Comment:
This is how transactions calculate it to save it in the TieredCache.

Maybe this info should also be somewhere in the TieredCache for addresses?
Maybe in "metadata for address"?
Please let me know your thoughts.

Reply:
**Update:** The main problem should be that a "selfdestruct" could be called for a contract which deletes its code so we could no longer recognize it as a contract here.
However do we not have the same problem for tx?
Does something more far reaching need to be changed?

Reply:
@recy21 answered here: https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1888#discussion_r1125705285

Reply:
The new issue to handle this is BIDS-1688.